monkey/monkey_island/cc/services/directory_file_storage_service.py

```diff
@@ -0,0 +1,54 @@
+import shutil
+from pathlib import Path
+from typing import BinaryIO
+
+from monkey_island.cc.server_utils.file_utils import create_secure_directory
+
+from . import IFileStorageService
+
+
+class DirectoryFileStorageService(IFileStorageService):
+    """
+    A implementation of IFileStorageService that reads and writes files from/to the local
+    filesystem.
+    """
+
+    def __init__(self, storage_directory: Path):
```

Comment:
Shouldn't we use `PurePath`?

Reply:
Pure paths "provide purely computational operations without I/O" [[1](https://docs.python.org/3.7/library/pathlib.html)]. Since we create this directory if it doesn't exist, I don't think a `PurePath` is appropriate.